components/wc-fin/src/components/context-group-picker/context-group-picker.tsx

```diff
@@ -204,31 +333,90 @@ export class ContextGroupPicker {
 				const urlParams = new URLSearchParams(window.location.search);
 				const contextGroupId = urlParams.get('contextGroupId');
 				if (contextGroupId !== undefined && contextGroupId !== null) {
-					await this.updateContextGroup(contextGroupId);
+					setTimeout(async () => {
+						await this.updateContextGroup(contextGroupId);
+					}, 1000);
 				}
 			}
 
 			if (this.contextGroupId === undefined) {
-				let options = await fin.me.getOptions();
-				let selectedContextGroup: string;
-
-				if (options.interop !== undefined && options.interop.currentContextGroup !== undefined) {
-					selectedContextGroup = options.interop.currentContextGroup;
-				} else if (
-					this.bindSelf === false &&
-					options.customData !== undefined &&
-					options.customData.selectedContextGroup !== undefined
-				) {
-					selectedContextGroup = options.customData.selectedContextGroup;
-				}
+				setTimeout(async () => {
+					let selectedContextGroup: string = await this.getCurrentContextGroup();
+					await this.updateContextGroup(selectedContextGroup);
+				}, 1000);
```

Comment:
To reduce all the individual timeouts you could create a reusable wrapper, eg

```
function delay(fn: (...args: any[]) => Promise<any>, delay: number = 1000) {
	return async (...args: any[]) => {
		setTimeout(async () => {
			await fn(...args);
		}, delay);
	};
}

const delayedUpdateContextGroup = delay(this.updateContextGroup);

if (this.contextGroupId === undefined) {
  let selectedContextGroup: string = await this.getCurrentContextGroup();
  await delayedUpdateContextGroup(selectedContextGroup);
}
```

Reply:
Hi @raycashmore, 

I've just pushed an update so that there isn't a scattering of setTimeouts and there is no need to wrap the function before calling it.

Moved the setTimeout logic into the functions themselves (so that the function itself can be updated as things change without impacting the calling code).

Also changed from setInterval for monitoring to using setTimeout and calling itself (so you never end up with more than one call on the event loop if the browser gets busy).

Continues to work although the context group selection is now a little slower as the calls are a bit slower because of the timeout.